Content.Shared/Construction/Conditions/NoUnstackableInTile.cs

```diff
@@ -0,0 +1,33 @@
+﻿using Content.Shared.Maps;
+using Content.Shared.Tag;
+using JetBrains.Annotations;
+using Robust.Shared.Map;
+
+namespace Content.Shared.Construction.Conditions
+{
+    [UsedImplicitly]
+    [DataDefinition]
+    public sealed class NoUnstackableInTile : IConstructionCondition
+    {
+        public bool Condition(EntityUid user, EntityCoordinates location, Direction direction)
+        {
+            var tagSystem = IoCManager.Resolve<IEntitySystemManager>().GetEntitySystem<TagSystem>();
+
+            foreach (var entity in location.GetEntitiesInTile(LookupFlags.Approximate | LookupFlags.Static | LookupFlags.Sundries))
```

Comment:
coordinates.getentitiesintile is obsolete and should be using entitylookupsystem, also see above about just making it a method call somewhere.

Reply:
Ah, okay, I just copied the `NoWindowsInTile` constraint.

Reply:
All I have access to here is a `EntityCoordinates location` which is passed in using `ConstructionSystem`. I reviewed the code called by `GetEntitiesInTile` and it seems to be a pretty clean lookup that converts `EntityCoordinates` into the required grid-id and bounding box.

I think that what you are asking for a quite in-depth refactor of the construction system.

Reply:
I found a query function and updated both the new stackable query and the old window one to the new lookup.